### PR TITLE
[CI] support k8s v1.24 in ci test

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
Signed-off-by: changzhen <changzhen5@huawei.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Support Kubernetes v1.24 in our CI E2E test.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/federatedresourcequota_test.go [hunk 1]
```diff
@@ -125,11 +125,10 @@ var _ = ginkgo.Describe("FederatedResourceQuota auto-provision testing", func()
 				clusterClient, err := util.NewClusterClientSet(clusterName, controlPlaneClient, nil)
 				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
 
-				gomega.Eventually(func(g gomega.Gomega) (bool, error) {
```
Comment (RainbowMango) [inline test/e2e/federatedresourcequota_test.go:128]: ?

Comment (XiShanYongYe-Chang) [reply test/e2e/federatedresourcequota_test.go]: Sorry, this is the code for the test, I will remove it.

**Comment by RainbowMango** [comment]
CI failed.

**Comment by XiShanYongYe-Chang** [comment]
`karmada-kube-controller-manager` is not running properly and is restarted six times. At the same time, `karmada-apiserver` is not working properly.

I suggest re-run the CI.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**